site2/docs/concepts-messaging.md

```diff
@@ -254,7 +254,7 @@ In the diagram below, only **Consumer A-0** is allowed to consume messages.
 
 ### Failover
 
-In *failover* mode, multiple consumers can attach to the same subscription. In failover mode, the broker selects the master consumer based on the priority level and the lexicographical sorting of a consumer name. If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting. If these two consumers have different priority levels, the broker selects the consumer with a higher priority level as the master consumer. The master consumer is initially the only one receiving messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line.
+In *failover* mode, multiple consumers can attach to the same subscription. For partitioned topic, in failover mode the broker selects the master consumer based on the priority level and the lexicographical sorting of a consumer name. If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting. If these two consumers have different priority levels, the broker selects the consumer with a higher priority level as the master consumer. The master consumer is initially the only one receiving messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line. For non-partitioned topic in failover mode the broker will pick consumer in the order they subscribe to the non partitioned topic.
 
 For partitioned topics, the broker assigns partitioned topics to the consumer with the highest priority level. If multiple consumers have the highest priority level, the broker evenly assigns topics to consumers with these consumers.
 
```

Comment:
Suggestions from language perspectives:

In *failover* mode, multiple consumers can attach to the same subscription. In failover mode, the broker selects the master consumer for the partitioned topic based on the priority level and the lexicographical sorting of a consumer name. If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting. If these two consumers have different priority levels, the broker selects the consumer with a higher priority level as the master consumer. Only the master consumer receives messages. When the master consumer is not connected, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line. 

The broker assigns partitioned topics to the consumer with the highest priority level. If multiple consumers have the same priority level, the broker assigns topics to consumers evenly. 

In failover mode, the broker selects the master consumer for a non-partitioned topic based on the order that they subscribe to the non-partitioned topic.

Reply:
Do the following two have some conflicts?
1. If multiple consumers have the same priority level, the broker assigns topics to consumers evenly."
2.  If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting.

Reply:
Thanks for the comment, yes it's conflicting description, I've updated the doc to reflect the latest way broker pick active consumer for topic in failover sub mode.